codalab/worker/docker_utils.py

```diff
@@ -261,6 +262,33 @@ def get_container_stats(container):
     return stats
 
 
+@wrap_exception('Unable to check Docker API for container')
+def get_container_stats_on_mac(container):
+    if container_exists(container):
+        return get_container_stats_helper_by_name(container.name)
+    else:
+        return '', ''
+
+
+@wrap_exception('Unable to check Docker API for container')
+def get_container_stats_helper_by_name(container_name):
+    try:
+        container_stats = client.containers.get(container_name).stats(stream=False)
+        cpu_used = int(container_stats['cpu_stats']['cpu_usage']['total_usage'])
+        total_cpu = int(container_stats['cpu_stats']['system_cpu_usage'])
+        cpu_usage = str(float(cpu_used / total_cpu))
+        memory_usage = str(
+            float(
+                container_stats['memory_stats']['usage'] / container_stats['memory_stats']['limit']
+            )
+        )
+        return cpu_usage, memory_usage
+    except KeyError:
```

Comment:
When would a KeyError occur?

Reply:
Oh I commented at the other place but forgot here. It's when the bundle that's being queried has finished/exited, some keys will be missing.

Reply:
Do you know why some keys would be missing? Is it a bug?

Reply:
It's not a bug. It's what docker apis return when called upon a finished/exited container.